libs/components/data-manager/testing/src/modules/data-manager/data-manager-toolbar-harness.ts

```diff
@@ -0,0 +1,259 @@
+import { HarnessPredicate } from '@angular/cdk/testing';
+import { SkyComponentHarness } from '@skyux/core/testing';
+import { SkyCheckboxHarness, SkyRadioGroupHarness } from '@skyux/forms/testing';
+import { SkyFilterButtonHarness, SkySortHarness } from '@skyux/lists/testing';
+import { SkySearchHarness } from '@skyux/lookup/testing';
+
+import { SkyDataManagerColumnPickerHarness } from './data-manager-column-picker-harness';
+import { SkyDataManagerToolbarHarnessFilters } from './data-manager-toolbar-harness-filters';
+import { SkyDataManagerToolbarLeftItemHarness } from './data-manager-toolbar-left-item-harness';
+import { SkyDataManagerToolbarLeftItemHarnessFilters } from './data-manager-toolbar-left-item-harness-filters';
+import { SkyDataManagerToolbarPrimaryItemHarness } from './data-manager-toolbar-primary-item-harness';
+import { SkyDataManagerToolbarPrimaryItemHarnessFilters } from './data-manager-toolbar-primary-item-harness-filters';
+import { SkyDataManagerToolbarRightItemHarness } from './data-manager-toolbar-right-item-harness';
+import { SkyDataManagerToolbarRightItemHarnessFilters } from './data-manager-toolbar-right-item-harness-filters';
+import { SkyDataManagerToolbarSectionHarness } from './data-manager-toolbar-section-harness';
+import { SkyDataManagerToolbarSectionHarnessFilters } from './data-manager-toolbar-section-harness-filters';
+
+/**
+ * Harness for interacting with a data manager toolbar component in tests.
+ */
+export class SkyDataManagerToolbarHarness extends SkyComponentHarness {
+  /**
+   * @internal
+   */
+  public static hostSelector = 'sky-data-manager-toolbar';
+
+  #documentRootLocator = this.documentRootLocatorFactory();
+
+  /**
+   * Gets a `HarnessPredicate` that can be used to search for a
+   * `SkyDataManagerToolbarHarness` that meets certain criteria.
+   */
+  public static with(
+    filters: SkyDataManagerToolbarHarnessFilters,
+  ): HarnessPredicate<SkyDataManagerToolbarHarness> {
+    return SkyDataManagerToolbarHarness.getDataSkyIdPredicate(filters);
+  }
+
+  /**
+   * Clicks the data manager clear all button, if it exists
+   */
+  public async clickClearAll(): Promise<void> {
+    const button = await this.locatorForOptional(
+      '.sky-data-manager-clear-all-btn',
+    )();
+
+    if (button === null) {
+      throw new Error('Unable to find the data manager clear all button.');
+    }
+
+    await button.click();
+  }
+
+  /**
+   * Clicks the data manager select all button, if it exists
+   */
+  public async clickSelectAll(): Promise<void> {
+    const button = await this.locatorForOptional(
+      '.sky-data-manager-select-all-btn',
+    )();
+
+    if (button === null) {
+      throw new Error('Unable to find the data manager select all button.');
```

Comment:
This is a button we manage right? What would be a situation where the button would not exist, can we add that context to the error so consumers know what they might need to do

Reply:
^same comment for a few other errors for buttons not found

Reply:
Updated